packages/sit-onyx/src/components/OnyxSwitch/OnyxSwitch.vue

```diff
@@ -0,0 +1,274 @@
+<script lang="ts" setup>
+import { OnyxIcon } from "@/index";
+import checkSmall from "@sit-onyx/icons/check-small.svg?raw";
+import xSmall from "@sit-onyx/icons/x-small.svg?raw";
+import type { OnyxSwitchProps } from "./types";
+import { computed } from "vue";
+const props = withDefaults(defineProps<OnyxSwitchProps>(), {
+  modelValue: false,
+  label: "Onyx Switch",
+  disabled: false,
+  required: false,
+  readonly: false,
+  invalid: false,
+});
+
+// TODO: Implement invalid disabled state.
+
+const emit = defineEmits<{
+  /** Emitted when the checked state changes. */
+  "update:modelValue": [value: boolean];
+}>();
+
+const isChecked = computed({
+  get: () => props.modelValue,
+  set: (value) => {
+    emit("update:modelValue", value);
+  },
+});
+</script>
+
+<template>
+  <label
+    tabindex="0"
+    class="onyx-switch"
+    :class="{ 'onyx-required-marker': props.required, 'onyx-optional-marker': !props.required }"
+  >
+    <input
+      v-model="isChecked"
+      class="onyx-switch__input"
+      :class="{
+        'onyx-switch__input--invalid': props.invalid,
```

Comment:
Please use the [`setCustomValidity`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setCustomValidity) function to set an error. Then you can use the `:invalid` pseudo-class.

Reply:
@JoCa96 I implemented the setCustomValidity function, but I'm note sure if I did it right?